src/modals/cpu-pinning/CpuPinningDataProvider.js

```diff
@@ -83,47 +111,93 @@ const getPlacementPolicyHostsIds = (vm) => {
   return hostsIds
 }
 
-const getCpus = (host) => {
-  const topology = host.cpu?.topology
-  if (topology) {
-    return topology.sockets * topology.cores * topology.threads
-  }
-  return 0
-}
-
-const mapVmToModel = (vm) => {
-  return {
-    id: vm.id,
-    name: vm.name,
-    cpuTopology: {
-      sockets: parseInt(getSockets(vm)),
-      cores: parseInt(getCores(vm)),
-      threads: parseInt(getThreads(vm)),
-    },
-    cpuPinningPolicy: vm.cpu_pinning_policy,
-    cpuPinnings: getPinnings(vm),
+const mapVmToPinnedEntity = (vm) => {
+  const pinned = new PinnedEntity(vm.id, vm.name)
+
+  const sockets = +getSockets(vm)
+  const cores = +getCores(vm)
+  const threads = +getThreads(vm)
+  const cpuToPinnedCpuMap = getCpuToPinnedCpuMapForVm(vm)
+
+  if (sockets && cores && threads) {
+    for (let s = 0; s < +sockets; s++) {
+      for (let c = 0; c < +cores; c++) {
+        for (let t = 0; t < +threads; t++) {
+          const cpuId = s * cores * threads + c * threads + t
+          pinned.cpuPinningTopology.add(s, c, cpuId, cpuToPinnedCpuMap.get(cpuId))
+        }
+      }
+    }
   }
+
+  pinned.cpuCount = sockets * cores * threads
+  pinned.cpuPinningPolicy = vm.cpu_pinning_policy
+  pinned.cpuPinningString = getCpuPinningString(vm)
+  return pinned
 }
 
-const mapHostToModel = (host) => {
-  return {
-    id: host.id,
-    name: host.name,
-    cpus: getCpus(host),
+const createCpuPinningTopology = (cpuUnits) => {
+  const cpuPinningTopology = new Topology()
+
+  cpuUnits?.host_cpu_unit?.forEach(cpuUnit => {
+    const socketId = +cpuUnit.socket_id
+    const coreId = +cpuUnit.core_id
+    const cpuId = +cpuUnit.cpu_id
+    const exclusivelyPinned = cpuUnit.vms?.vm.some(vm => CpuPinningPolicy.isExclusive(vm.cpu_pinning_policy))
+    const pinnedEntities = (cpuUnit.vms?.vm.map(vm => vm.name) || []).sort()
+    if (cpuUnit.runs_vdsm) {
+      pinnedEntities.unshift('VDSM')
```

Comment:
what happens if a VM is named `VDSM` :) ?

Reply:
yes, that might be problem :-) Maybe we can change it to VDS Manager (no VM can have space in the name). I am only not sure if we should localize it or not. @ahadas WDYT?

Reply:
I think it's ok to keep it as VDSM - without localization